rfcs/text/0408-versioning-types.md

```diff
@@ -0,0 +1,344 @@
+1.  If a property in `A` is defined as an array, then it's also defined as an array in `B` (and therefore the same for direct references)
+1.  If a property in `A` is defined as an array with a `minItems`, then there is a `minItems` constraint on the respective property in `B` that is equal to or larger to that in `A`
+1.  If a property in `A` is defined as an array with a `maxItems`, then there is a `maxItems` constraint on the respective property in `B` that is less than or equal to that in `B`
+1.  Each property type referenced in `A` is compatible with the respective property type referenced in `B`. Where respective is defined by having the same base URI (property key)
+1.  Each URI in the `required` list of `A` is also in the `required` list of `B`
+1.  Each key in the `links` of `A` is also in the `links` of `B`
+1.  If a link in `links` of `A` is defined as an array, then it's also defined as an array in `B` (and therefore the same for direct references)
+1.  If a link in `links` of `A` is defined as an array with a `minItems`, then there is a `minItems` constraint on the respective link in `B` that is equal to or larger to that in `A`
+1.  If a link in `links` of `A` is defined as an array with a `maxItems`, then there is a `maxItems` constraint on the respective link in `B` that is less than or equal to that in `B`
+1.  If a link in `links` of `A` is defined as an array with `ordered` as `true`, then there is an `ordered` as `true` on the respective link in `B`
+1.  Each URI in the `requiredLinks` list of `A` is also in the `requiredLinks` list of `B`
+
+### Link Types
+
+Link Types are only able to be changed through _semantic annotations_ and as such, any Link Type `X`, with the same Base URI as another Link Type `Y` (i.e. a different version of the same link type), are _compatible_.
+
+## When to Check Compatibility of Types
+
+Block schemas, and other uses of types (for example graph service methods), can request data from the embedding application by referencing types. Not relying on the version to ensure compatibility means that if a block, or data (associated with a versioned type) in an embedding application, is "outdated", the compatibility can be checked by comparing the two schemas. Data blobs can continue to be validated against the relevant schema.
+
+This should allow a bit more flexibility within the system when it comes to compatibility of data requirements.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+- This proposal makes the active decision to not embed compatibility information within the version identifier. This contrasts with approaches like semantic versioning, which is explored more in the [Rationale and alternatives](#rationale-and-alternatives) section.
+- The versioning approach does not encapsulate information about the scale of the change, fixing a typo causes a new version in the same way that completely rewriting the constraints does.
```

Comment:
```suggestion
- The versioning approach does not encapsulate information about the scale of the change: fixing a typo causes a new version in the same way that completely rewriting the constraints does (assuming changes are not batched before a new version URI is created).
```

Reply:
I don't think this makes it clearer, I actually think it's more confusing.

Reply:
The point I'm trying to make is that what 'causes a new version' is entirely up to whoever is managing the creation of these types. This might be read as implying that fixing a typo _must_ cause a new version.

Reply:
Right but fixing a typo does mean you _must_ have a new version. Whether or not you include other changes within that doesn't change it.

I explain this explicitly, including something similar to your recommendation, above

> ## What is a "version" of a Type
>
> Any published (accessible) iteration of a Type can be considered a "version" of the Type. Once a specific iteration of a type is made accessible via a URI, it should be considered immutable.
> 
> - This means that any change made to the Type (for example changing the description, adding a field, etc) constitutes reason for a new version.
> - **If multiple changes are made prior to making it accessible (i.e. batching the changes), then a separate version is not needed for each individual change, but they can all be 'released' together.**

Reply:
Right - my recommendation was a callback to that, to reinforce the point that you can fix a typo and then leave that draft version somewhere without creating it. 

It's not important - resolving this